crates/polars-plan/src/plans/optimizer/simplify_expr.rs

```diff
@@ -440,6 +440,63 @@ impl OptimizationRule for SimplifyExprRule {
         let expr = expr_arena.get(expr_node).clone();
 
         let out = match &expr {
+            // drop_nulls().len() -> null_count()
+            // drop_nulls().count() -> null_count()
+            AExpr::Agg(IRAggExpr::Count(input, _)) => {
+                let input_expr = expr_arena.get(*input);
+                match input_expr {
+                    AExpr::Function {
+                        input,
+                        function: FunctionExpr::DropNulls,
+                        options: _,
+                    } => {
+                        let expr_ir = ExprIR::from_node(expr_node, expr_arena);
+                        let expr_output_name = expr_ir.output_name();
+                        let inner_minus_exp = AExpr::BinaryExpr {
+                            op: Operator::Minus,
+                            right: expr_arena.add(make_null_count_expr!(input)),
+                            left: expr_arena.add(AExpr::Len),
```

Comment:
This is under `drop_nulls().count()` - it should just be the null count expr instead of `len - null_count`?